File: app/src/main/java/com/example/TibiaTools/APISERVER/TibiaAPIServer.java
package com.example.TibiaTools.APISERVER;

import com.example.TibiaTools.APISERVER.models.APIBoostableBosses;
import com.example.TibiaTools.APISERVER.models.APICriatures;
import com.example.TibiaTools.APISERVER.models.APICriaturesInformation;
import com.example.TibiaTools.APISERVER.models.ApiHighScores;
import com.example.TibiaTools.APISERVER.models.ApiHouses;
import com.example.TibiaTools.APISERVER.models.ApiHousesInformation;
import com.example.TibiaTools.APISERVER.models.ApiNews;
import com.example.TibiaTools.APISERVER.models.ApiNewsTicker;
import com.example.TibiaTools.APISERVER.models.ApiSpells;
import com.example.TibiaTools.APISERVER.models.ApiSpellsInformation;
import com.example.TibiaTools.APISERVER.models.CharactersInformation.APIServicesTibia;
import com.example.TibiaTools.APISERVER.models.GuildInformation.ApiGuilds;
import com.example.TibiaTools.APISERVER.models.GuildInformation.ApiGuildsName;
import com.example.TibiaTools.APISERVER.models.Worlds.DataWords;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface TibiaAPIServer {
    @GET("/v4/character/{name}")
    Call<APIServicesTibia> getPersonajes(@Path("name") String name);

    @GET ("/v4/worlds")
    Call<DataWords> getWorlds();

    @GET ("/v4/creatures")
    Call<APICriatures> getCreature();

    @GET("/v4/boostablebosses")
    Call<APIBoostableBosses> getBoostableBosses();

    @GET("{race}")
    Call<APICriaturesInformation> getCriatureInformation(@Path("race") String race);

    @GET("/v4/guilds/{world}")
    Call<ApiGuilds> getGuildsInformation(@Path("world") String world);

    @GET("/v4/guild/{name}")
    Call<ApiGuildsName> getGuildsInformationName(@Path("name") String name);

    @GET("spells")
    Call<ApiSpells> getSpells();

    @GET("{spell}")
    Call<ApiSpellsInformation> getSpellInformation(@Path("spell") String spell);

    @GET("https://api.tibialabs.com/v2/rashid")
    Call<String> getRashidLocalitation();

    @GET("/v4/news/latest")
    Call<ApiNews> getNewsLatest();

    @GET("/v4/news/newsticker")
    Call<ApiNewsTicker> getNewsTickers();

    @GET("/v4/highscores/{world}/{category}/{vocation}")
    Call<ApiHighScores> getHighScoreInformation(@Path("world") String world, @Path("category") String category, @Path("vocation") String vocation);

    @GET("houses/{world}/{town}")
    Call<ApiHouses> getHousesInformation(@Path("world") String world, @Path("town") String town);
    @GET("house/{world}/{house_id}")
    Call<ApiHousesInformation> getHouseInformation(@Path("world") String world, @Path("house_id") String house_id);
}

File: app/src/main/java/com/example/TibiaTools/data/retrofit/TibiaAPIServer.kt
package com.example.TibiaTools.data.retrofit

import com.example.TibiaTools.APISERVER.models.APIBoostableBosses
import com.example.TibiaTools.APISERVER.models.APICriatures
import com.example.TibiaTools.APISERVER.models.APICriaturesInformation
import com.example.TibiaTools.APISERVER.models.ApiHighScores
import com.example.TibiaTools.APISERVER.models.ApiHouses
import com.example.TibiaTools.APISERVER.models.ApiHousesInformation
import com.example.TibiaTools.APISERVER.models.ApiNews
import com.example.TibiaTools.APISERVER.models.ApiNewsTicker
import com.example.TibiaTools.APISERVER.models.ApiSpells
import com.example.TibiaTools.APISERVER.models.ApiSpellsInformation
import com.example.TibiaTools.APISERVER.models.CharactersInformation.APIServicesTibia
import com.example.TibiaTools.APISERVER.models.GuildInformation.ApiGuilds
import com.example.TibiaTools.APISERVER.models.GuildInformation.ApiGuildsName
import com.example.TibiaTools.APISERVER.models.Worlds.DataWords
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Url

interface TibiaAPIServer {
    @GET("/v4/character/{name}")
    suspend fun getPersonajes(@Path("name") name: String): APIServicesTibia

    @GET("/v4/worlds")
    suspend fun getWorlds(): DataWords

    @GET("/v4/creatures")
    suspend fun getCreature(): APICriatures

    @GET("/v4/boostablebosses")
    suspend fun getBoostableBosses(): APIBoostableBosses

    @GET("/v4/creature/{race}")
    suspend fun getCriatureInformation(@Path("race") race: String): APICriaturesInformation

    @GET("/v4/guilds/{world}")
    suspend fun getGuildsInformation(@Path("world") world: String): ApiGuilds

    @GET("/v4/guild/{name}")
    suspend fun getGuildsInformationName(@Path("name") name: String): ApiGuildsName

    @GET("/v4/spells")
    suspend fun getSpells(): ApiSpells

    @GET("/v4/spell/{spell}")
    suspend fun getSpellInformation(@Path("spell") spell: String): ApiSpellsInformation

    @GET("https://api.tibialabs.com/v2/rashid")
    suspend fun getRashidLocalitation(): String

    @GET("/v4/news/latest")
    suspend fun getNewsLatest(): ApiNews

    @GET("/v4/news/newsticker")
    suspend fun getNewsTickers(): ApiNewsTicker

    @GET("/v4/highscores/{world}/{category}/{vocation}")
    suspend fun getHighScoreInformation(
        @Path("world") world: String,
        @Path("category") category: String,
        @Path("vocation") vocation: String
    ): ApiHighScores

    @GET("/v4/houses/{world}/{town}")
    suspend fun getHousesInformation(
        @Path("world") world: String,
        @Path("town") town: String
    ): ApiHouses

    @GET("/v4/house/{world}/{house_id}")
    suspend fun getHouseInformation(
        @Path("world") world: String,
        @Path("house_id") houseId: String
    ): ApiHousesInformation
}
